roles/nix-community-cache.nix

```
@@ -1,51 +1,17 @@
-{ config, pkgs, ... }:
+{ config, pkgs, lib, ... }:
 let
-  postBuildHook = pkgs.writeScript "post-build-hook.sh" ''
-    #!${pkgs.runtimeShell}
-    export PATH=$PATH:${pkgs.nixFlakes}/bin
-    exec ${pkgs.cachix}/bin/cachix -c /var/lib/post-build-hook/nix-community-cachix.dhall push nix-community $OUT_PATHS
-  '';
-
-  sockPath = "/run/post-build-hook.sock";
-
-  queueBuildHook = pkgs.writeScript "post-build-hook.sh" ''
-    ${pkgs.queued-build-hook}/bin/queued-build-hook queue --socket ${sockPath}
-  '';
-
-  sources = import ../nix/sources.nix;
+  configFile = "/var/lib/post-build-hook/nix-community-cachix.dhall";
 
 in
 {
-
-  nixpkgs.overlays = [
-    (self: super: {
-      queued-build-hook = (import sources.queued-build-hook { pkgs = super; });
-    })
-  ];
-
-  systemd.sockets.queued-build-hook = {
-    description = "Post-build-hook socket";
-    wantedBy = [ "sockets.target" ];
-    socketConfig = {
-      ListenStream = sockPath;
-      SocketUser = "root";
-      SocketMode = "0600";
-    };
-  };
-
-  systemd.services.queued-build-hook = {
-    description = "Post-build-hook service";
+  systemd.services.cachix-watch-store = {
+    description = "Cachix store watcher service";
     wantedBy = [ "multi-user.target" ];
-    after = [ "network.target" "queued-build-hook.socket" ];
-    requires = [ "queued-build-hook.socket" ];
+    after = [ "network.target" ];
+    path = [ pkgs.nix ];
```

Let's use the same version of nix everywhere:
```suggestion
    path = [ config.nix.package ];
```